### PR TITLE
Apply changes for the upcoming build infrastructure improvements (1.x)

### PR DESCRIPTION
**Pull request by warpech**
point 8 from CONTRIBUTING.md assumes changes from: Starcounter/AdminTrack#438

point 4 from CONTRIBUTING.md assumes changes from: Starcounter/AdminTrack#435

@joozek78, @ubbeK please review.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,29 +1,22 @@
...
-10. If all tests pass, create a PR to level1 (remember to bump `CHANGELOG.md`)
-11. You don't need a reviewer for this PR. Simply do something similar to what @warpech did in https://github.com/Starcounter/level1/pull/4420 and give the link to the PR in StarcounterClientFiles where it was reviewed in more detail.
-12. Merge the PR to `v2.3` and then merge `v2.3` to `develop`. Take proper steps to ensure that no unintended files were overwritten in `develop`. If yes, solve the conflict.
+4. When you push to a feature branch of StarcounterClientFiles, TeamCity will run a integration test of this branch with KitchenSink, WebsiteProvider and CompositionProvider. Make sure these tests are passing.
+5. Make a PR for someone to review the complete set of changes.
+6. Still in the feature branch, release new version (1.x) of StarcounterClientFiles on GitHub. Remember to write the release notes.
+7. Still in the feature branch, publish a package of this version to App Warehouse (built with Starcounter 2.3)
+8. Merge the PR to StarcounterClientFiles branch `1.x`.
+   - at this point, a new build of Level1 is started
```
Comment (ubbeK) [inline CONTRIBUTING.md]: I am uncertain if this specific part is true.

@joozek78 Isn't it only `Level0` that has the automatic trigger on PR:s?

Comment (warpech) [reply CONTRIBUTING.md]: Actually, I want to know if either of the following is true:

1. a new build of Level1 is started when a commit is pushed to StarcounterClientFiles `1.x` / `2.x` branch
2. or: a nightly build of Level1 will include changes from pushed to StarcounterClientFiles `1.x` / `2.x` even if no changes were detected in the Level1 repo 

I suppose that 2 is true and 1 is not.

If you can confirm that, I will change the text here to reflect that.

Comment (ubbeK) [reply CONTRIBUTING.md]: 1. Yes, I belive it is, if you mean `daily windows build` by `Level1 build`. https://github.com/Starcounter/AdminTrack/issues/438#issuecomment-363460042
2. The `nightly windows builds` are only triggered at night if there are changes in any of the 11 repositories "default" branch where `StarcounterClientFiles` is one of them. So this step will not be a problem for you.

Comment (warpech) [reply CONTRIBUTING.md]: Perfetto!

Comment (warpech) [reply CONTRIBUTING.md]: Added a clarification about that

**Review by ubbeK — APPROVED** [review]

**Pull request merged**